csFastFloat/FastDoubleParser.cs

```diff
@@ -90,7 +90,7 @@ public static unsafe bool TryParseDouble(string s, out double result, NumberStyl
     /// <returns></returns>
     public static unsafe bool TryParseDouble(string s, out int characters_consumed, out double result, NumberStyles styles = NumberStyles.Float, char decimal_separator = '.')
     {
-      if (string.IsNullOrEmpty(s))
+      if (s == null)
```

Comment:
This change is OK, but either flip the condition to `if (s != null)` (for the branch preditor) or (just came into my mind) take a sligth different approach:
```c#
return TryParseDouble(s.AsSpan(), out characters_consumed, out result, ...);
```
so to forward to the span-based version.
In the case of `null` this results in a empty span, so that length will be 0 and algorithm returns (without throwing an null-ref exception).

Reply:
If followed your second idea, fowarding to span-based.  I haven't noticed before that the asSpan function was an extension on string? !  This is really intersting thanks.